desc/objectives/_qs.py

```diff
@@ -143,6 +143,7 @@ def build(self, eq=None, use_jit=True, verbose=1):
             helicity=self.helicity,
             NFP=self._transforms["B"].basis.NFP,
         )
+        self._idx = np.where(self._idx)[0]
```

Comment:
Can we fix the issue in `ptolemy_linear_transform` rather than in the Boozer objective? 
In theory this util function could be used in other places, and then the same issue would arise. So better to fix the root cause.